src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -3,580 +3,381 @@
 
 using System;
 using System.Diagnostics;
-
+using System.Runtime.InteropServices;
 using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Win32.SafeHandles;
 using Xunit;
 using System.Text;
 
 namespace System.Diagnostics.ProcessTests
 {
-    public partial class ProcessTest
+    public partial class ProcessTest : IDisposable
     {
-        static ProcessTest()
-        {
-            TryKillExistingTestProcesses();
-        }
 
-        private static void TryKillExistingTestProcesses()
+        private const string s_ProcessName = "ProcessTest_ConsoleApp.exe";
+        private static bool s_Process_EnableRaiseEvents_isExitedEventHandlerCalled = false;
+        private Process s_process;
+
+        public ProcessTest()
         {
-            try
-            {
-                foreach (Process p in Process.GetProcessesByName(s_ProcessName.Replace(".exe", "")))
-                {
-                    p.Kill();
-                }
-            }
-            catch { }; // do nothing 
+            s_process = CreateProcessInfinite();
```

Comment:
The "infinite" process only runs for 30 seconds.  If we're using one such process for the whole ProcessTest suite, is there any concern that the process tests could end up taking longer than that on a given run and end up failing as a result?